staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go

```diff
@@ -421,6 +421,33 @@ func (s *GenericAPIServer) PrepareRun() preparedGenericAPIServer {
 
 // Run spawns the secure http server. It only returns if stopCh is closed
 // or the secure port cannot be listened on initially.
+// This is the diagram of what channels/signals are dependent on each other:
+//
+//                                  stopCh
+//                                    |
+//           ---------------------------------------------------------
+//           |                                                       |
+//    ShutdownInitiated (shutdownInitiatedCh)                        |
+//           |                                                       |
+// (ShutdownDelayDuration)                                    (PreShutdownHooks)
+//           |                                                       |
+//  AfterShutdownDelayDuration (delayedStopCh)           preShutdownHooksHasStoppedCh
+//           |                                                       |
+//           |----------------------------------                     |
+//           |                                  |                    |
+//           |                       (HandlerChainWaitGroup::Wait)   |
```

Comment:
hehe, now is clear

Reply:
i was working on a hackmd a while back - https://hackmd.io/yjCkQzImQs-_qC2yAW9EuA, let me know if you can access it.

Reply:
it needs some more work though

Reply:
Yes - I can access. We can link it here eventually - for now we can leave what I added here.